src/streamlink_cli/main.py

```diff
@@ -650,20 +651,32 @@ def setup_args(parser: argparse.ArgumentParser, config_files: List[Path] = None,
 def setup_config_args(parser, ignore_unknown=False):
     config_files = []
 
-    if streamlink and args.url:
-        with ignored(NoPluginError):
-            plugin = streamlink.resolve_url(args.url)
-            config_files += [path.with_name(f"{path.name}.{plugin.module}") for path in CONFIG_FILES]
-
     if args.config:
         # We want the config specified last to get highest priority
-        config_files += map(lambda path: Path(path).expanduser(), reversed(args.config))
+        for config_file in map(lambda path: Path(path).expanduser(), reversed(args.config)):
+            if config_file.is_file():
+                config_files.append(config_file)
     else:
         # Only load first available default config
         for config_file in filter(lambda path: path.is_file(), CONFIG_FILES):
+            if type(config_file) is DeprecatedPath:
+                log.info(f"Loaded config from deprecated path, see CLI docs for how to migrate: {config_file}")
             config_files.append(config_file)
             break
 
+    if streamlink and args.url:
+        # Only load first available plugin config
+        with ignored(NoPluginError):
+            plugin = streamlink.resolve_url(args.url)
+            for config_file in CONFIG_FILES:
+                config_file = config_file.with_name(f"{config_file.name}.{plugin.module}")
+                if not config_file.is_file():
+                    continue
+                if type(config_file) is DeprecatedPath:
+                    log.info(f"Loaded plugin config from deprecated path, see CLI docs for how to migrate: {config_file}")
+                config_files.append(config_file)
+                break
+
```

Comment:
There is a small change in logic here. Previously it was trying to load multiple plugin specific config files, eg. `~/.config/streamlink/config.twitch` and `~/.streamlinkrc.twitch`, but since only the first non-plugin-specific default config file gets loaded (see above), plugin configs should have the same logic.

I'm also not sure if we should disable plugin config loading when `--config` is set or if we should add something like `--no-plugin-config` for disabling loading plugin configs.